### PR TITLE
kill: migrate to clap

### LINKED ISSUE #2122: kill: move from getopts to clap
Code is in:
https://github.com/uutils/coreutils/blob/master/src/uu/kill/src/kill.rs

Tests (definitely need some):
https://github.com/uutils/coreutils/blob/master/tests/by-util/test_kill.rs 

As example:
https://github.com/uutils/coreutils/commit/9081e120d6b866c94679d9e29c7bdb17ccccb4d0

### PR DESCRIPTION
**Pull request by nthery**
- Migrate to clap.
- Change default signal to SIGTERM.
- Add integration tests.

Fixes #2122.

### CODE REVIEW

File: src/uu/kill/src/kill.rs [hunk 1]
```diff
@@ -10,14 +10,19 @@
...
 use std::io::Error;
 use uucore::signals::ALL_SIGNALS;
 use uucore::InvalidEncodingHandling;
 
-static SYNTAX: &str = "[options] <pid> [...]";
-static SUMMARY: &str = "";
-static LONG_HELP: &str = "";
+static VERSION: &str = env!("CARGO_PKG_VERSION");
+static ABOUT: &str = "Send signal to processes or list information about signals.";
+static OPT_PIDS_OR_SIGNALS: &str = "PID";
```
Comment (sylvestre) [inline src/uu/kill/src/kill.rs:21]: please use this syntax to declare the options

https://github.com/uutils/coreutils/search?q=%22pub+mod+options%22

File: src/uu/kill/src/kill.rs [hunk 2]
```diff
@@ -33,41 +38,69 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
...
-            "list all signal names, or convert one to a name",
-            "LIST",
+
+    let usage = format!("{} [OPTIONS]... PID...", executable!());
+    let matches = App::new(executable!())
+        .version(VERSION)
+        .about(ABOUT)
+        .usage(&usage[..])
+        .arg(
+            Arg::with_name(OPT_LIST)
```
Comment (sylvestre) [inline src/uu/kill/src/kill.rs:49]: and use it here :)

**Review by sylvestre — CHANGES_REQUESTED** [review]
well done for the tests :)
just some minor changes

File: src/uu/kill/src/kill.rs (hunk 2 above)
Comment (nthery) [reply src/uu/kill/src/kill.rs:49]: Ok.  Thx for the review.

File: src/uu/kill/src/kill.rs [hunk 3]
```diff
@@ -33,41 +41,69 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
...
+
+    let usage = format!("{} [OPTIONS]... PID...", executable!());
+    let matches = App::new(executable!())
+        .version(VERSION)
+        .about(ABOUT)
+        .usage(&usage[..])
+        .arg(
+            Arg::with_name(options::LIST)
+                .short("l")
+                .long("list")
```
Comment (sylvestre) [inline src/uu/kill/src/kill.rs:54]:
```suggestion
                .long(options::LIST)
```

File: src/uu/kill/src/kill.rs [hunk 4]
```diff
@@ -33,41 +41,69 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
...
+                .help("Lists signals")
+                .conflicts_with(options::TABLE_NEW)
+                .conflicts_with(options::TABLE_OLD),
+        )
+        .arg(
+            Arg::with_name(options::TABLE_NEW)
+                .short("t")
+                .long("table")
```
Comment (sylvestre) [inline src/uu/kill/src/kill.rs:62]:
```suggestion
                .long(options::TABLE_NEW)
```

File: src/uu/kill/src/kill.rs [hunk 5]
```diff
@@ -33,41 +41,69 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
...
+                .help("Lists table of signals"),
+        )
+        .arg(Arg::with_name(options::TABLE_OLD).short("L").hidden(true))
+        .arg(
+            Arg::with_name(options::SIGNAL)
+                .short("s")
+                .long("signal")
```
Comment (sylvestre) [inline src/uu/kill/src/kill.rs:69]:
```suggestion
                .long(options::SIGNAL)
```

File: src/uu/kill/src/kill.rs [hunk 6]
```diff
@@ -33,41 +41,69 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
...
+    } else if matches.is_present(options::LIST) {
         Mode::List
     } else {
         Mode::Kill
     };
 
+    let pids_or_signals: Vec<String> = matches
+        .values_of(options::PIDS_OR_SIGNALS)
+        .map(|v| v.map(ToString::to_string).collect())
+        .unwrap_or_default();
```
Comment (sylvestre) [inline src/uu/kill/src/kill.rs:91]:
```suggestion
        .unwrap_or_default();

```

**Review by sylvestre — CHANGES_REQUESTED** [review]
minor changes :)

**Comment by sylvestre** [comment]
Thanks to your patch, we are now about 70% code coverage! bravo :)
```Successful in 1s — 70.09% (+0.82%) ```

**Pull request merged**